Task2/TrainingManagementSystem/Training.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace TrainingManagementSystem
+{
+    internal class Training : TrainingComponent
+    {
+        List<TrainingComponent> trainingsAndExercises = new List<TrainingComponent>();
+
+        public Training()
+        {
+            trainingsAndExercises = new List<TrainingComponent>();
+        }
+
+        public List<TrainingComponent> TrainingsAndExercises => trainingsAndExercises;
+
+        public bool Add(TrainingComponent trainingComponent)
+        {
+            if (trainingComponent.GetType() != GetType() && TrainingsAndExercises.Contains(trainingComponent))
+            {
+                trainingsAndExercises.Add(trainingComponent);
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool IsPractical()
+        {
+            foreach (var trainingComponent in TrainingsAndExercises)
+            {
+                if (trainingComponent.GetType() == typeof(Lecture))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public Training Clone()
+        {
+            Training clone = new Training{ Description = this.Description };
+
+            foreach (TrainingComponent trainingComponent in TrainingsAndExercises)
+            {
```

Comment:
Implement Clone interface for Lecture and PracticalLesson and remove if else statement